saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/authentication/OpenSamlAuthenticationProvider.java

```diff
@@ -512,6 +566,12 @@ private Saml2ResponseValidatorResult validateResponse(Saml2AuthenticationToken t
 	}
 
 	private List<Assertion> decryptAssertions(Decrypter decrypter, Response response) {
+		Consumer<Response> assertionDecrypter = getAssertionDecrypter();
+		if (assertionDecrypter != null) {
```

Comment:
Instead of checking for null, could we instead set a default value to `assertionDecrypter` that performs the default operation?

Reply:
I committed a new change to match with your suggestion. To do it i had to create a class variable for the Decrypter since it's now being used from the 2 new consumers.